### ISSUE TITLE
Porting to Plasma 6?

### ISSUE THREAD

**Issue by fenglengshun** [psifidotos/applet-window-buttons#209]
As per [Nate's article](https://pointieststick.com/2023/10/24/its-time-to-port-your-widgets-to-plasma-6/). Does this already work in Plasma 6? Is it expected to be easy to port? Is there possibility of improving it so that you can install it from Get New Widgets just fine? Is there already a separate repo that someone/a group actively maintains?

We've been hacking it for a while now, but I hope this widget don't die and the pain points we've had could finally be fixed.

**Comment by ShalokShalom** [comment]
I found it can cause the panel to stay absent in "windows can cover" mode and Wayland. 
The good thing is, that "windows can cover" gets deprecated and rewritten from scratch. 

> Is there possibility of improving it so that you can install it from Get New Widgets just fine?

Oh, so the kdecorations issue does also happen for you?

![Screenshot_20231031_000439](https://github.com/psifidotos/applet-window-buttons/assets/6344099/2a30bc5f-b892-4664-90a7-a9ab63a7b6d4)

Is there already an issue open?

I would be up for testing. 

I would also suggest to make this an official part of the KDE project.
As we might find more contributors this way.

**Comment by fenglengshun** [comment]
> Is there already an issue open?

@ShalokShalom I believe that would be issue #193, though if you're on Fedora or Arch, the dnf and AUR packages are built correctly to solve that issue.

**Comment by ShalokShalom** [comment]
I was on Arch, and I recently switched to NixOS. 

I will forward this to the packager there. 🙂 
Thanks

**Comment by luisbocanegra** [comment]
> As per [Nate's article](https://pointieststick.com/2023/10/24/its-time-to-port-your-widgets-to-plasma-6/). Does this already work in Plasma 6? 

Nope, fails to build. Will send build log later

> Is it expected to be easy to port? 

That would depend on how much plasma 6 changed to the parts this plugin uses

>Is there possibility of improving it so that you can install it from Get New Widgets just fine? 

Only way I think would be making it qml only and removing the features that require C++. 
I am currently trying to do that so I can use plasma 6 (right now is just max/minimize close buttons, static svg elements) and in case this never gets ported at least will have that

> Is there already a separate repo that someone/a group actively maintains?

Looking at forks at least, doesn't look like anyone is doing that

**Comment by pallaswept** [comment]
> (right now is just max/minimize close buttons, static svg elements) and in case this never gets ported at least will have that

Would you be so kind as to share that? I don't use titlebars at all, so this applet is my only way of managing windows with the mouse (and sometimes I can only use the mouse, I'm disabled), and when plasma 6 is released in a few months, I'm totally screwed without this applet. If you've got something that works on plasma 6, I'd really appreciate having it, too.

**Comment by luisbocanegra** [comment]
I uploaded it here https://github.com/luisbocanegra/plasma-simple-window-buttons

It is still in the same state as before because I haven't had time to work on it. 

**Currently only appears on maximized windows because that's how I use `applet-window-buttons`**

**Comment by pallaswept** [comment]
> I uploaded it here https://github.com/luisbocanegra/plasma-simple-window-buttons
> 
> It is still in the same state as before because I haven't had time to work on it.
> 
> **Currently only appears on maximized windows because that's how I use `applet-window-buttons`**

You absolute legend, thank you so much ❤️

**Comment by moodyhunter** [comment]
Got a semi-working Qt6/KF6/plasma6 port on my fork, https://github.com/moodyhunter/applet-window-buttons6/commit/4eae4e82f90c7b184e04ffa3ae347c64e0292f1d
I'll experiment and test for days and possibly (if it doesn't break that much) PR it.

**Comment by pallaswept** [comment]
> Got a semi-working Qt6/KF6/plasma6 port on my fork, [moodyhunter@4eae4e8](https://github.com/moodyhunter/applet-window-buttons6/commit/4eae4e82f90c7b184e04ffa3ae347c64e0292f1d) I'll experiment and test for days and possibly (if it doesn't break that much) PR it.

Already has 46 forks lol I think it's safe to say we're all enthusiastic :D Thanks dude

**Comment by jvyden** [comment]
@moodyhunter I've tried installing both your applet-window-title6 and applet-window-buttons6 forks, but they both still seem to show the 'unsupported widget' warning, and I can't add them.

![image](https://github.com/psifidotos/applet-window-buttons/assets/40577357/b376eaab-bcf1-43e7-a907-28b04e5a0070)

I still seem to be able to run them in `plasmoidviewer`, though:

![image](https://github.com/psifidotos/applet-window-buttons/assets/40577357/0c68e325-cb6b-4ac4-91f8-dad124f1bd58)

**Comment by luisbocanegra** [comment]
> @moodyhunter I've tried installing both your applet-window-title6 and applet-window-buttons6 forks, but they both still seem to show the 'unsupported widget' warning, and I can't add them.

Probably it needs `"X-Plasma-API-Minimum-Version": "6.0"` try adding it 

https://invent.kde.org/plasma/plasma-workspace/-/blob/master/applets/brightness/package/metadata.json?ref_type=heads

**Comment by moodyhunter** [comment]
True, I've added X-Plasma-API-Minimum-Version for both applets, try it now.

kwin 6 breaks my production machine and I'm forced to downgrade back to 5. :(

**Comment by VesperLlama** [comment]
@moodyhunter I installed your applet-window-buttons6 fork. But it's failing to start and giving the following error 
``module "org.kde.appletdecoration" is not installed``

I am on Arch with Plasma 6 Beta 1 and have kdecoration installed. Did I miss something?

**Comment by pallaswept** [comment]
FWIW, I actually swapped out the window buttons addon for this beauty https://github.com/luisbocanegra/plasma-panel-spacer-extended which already works on Plasma 6 too, so I'm ready for v6 now :)

**Comment by moodyhunter** [comment]
> @moodyhunter I installed your applet-window-buttons6 fork. But it's failing to start and giving the following error `module "org.kde.appletdecoration" is not installed`
> 
> I am on Arch with Plasma 6 Beta 1 and have kdecoration installed. Did I miss something?

Can you check if `/usr/lib/qt6/qml/org/kde/appletdecoration` exists? it's owned by this package.

**Comment by luisbocanegra** [comment]
I installed https://github.com/moodyhunter/applet-window-buttons6 today in up-to-date Neon unstable and don't have the error @VesperLlama had

**Comment by VesperLlama** [comment]
> Can you check if `/usr/lib/qt6/qml/org/kde/appletdecoration` exists? it's owned by this package.

I didn't have the folder in that location. It was in `/usr/lib/qml/org/kde/appletdecoration` on my system. I moved it to `/usr/lib/qt6/qml/org/kde/appletdecoration` and now it works fine.

Thanks for the help!

But what could have caused this problem?

**Comment by IF-tiger** [comment]
> 我安装了你的 applet-window-buttons6 fork。但它无法启动并给出以下错误 `module "org.kde.appletdecoration" is not installed`
> 
> 我在 Arch 上使用 Plasma 6 Beta 1 并安装了 kdecoration。我错过了什么吗？

I installed successfully on my opensuse device 
https://github.com/IT-Tagore/applet-window-buttons6/blob/master/README.md

**Comment by matiux** [comment]
@moodyhunter I have same problem with your package:

```
file:///usr/share/plasma/plasmoids/org.kde.windowbuttons/contents/ui/main.qml:26:1: module "org.kde.appletdecoration" is not installed
```

but `/usr/lib/qt6/qml/org/kde/appletdecoration` exists on my Arch Linux:
```
ls /usr/lib/qt6/qml/org/kde/appletdecoration
AuroraeButton.qml  libappletdecorationplugin.so  qmldir
```

**Comment by moodyhunter** [comment]
@matiux Hi, can you provide the plasmashell's output? i.e. the output when running `plasmashell` directly in your terminal.

**Comment by matiux** [comment]
@moodyhunter after restating now works... I don't know why :/

**Comment by witt-bit** [comment]
> @matiux Hi, can you provide the plasmashell's output? i.e. the output when running `plasmashell` directly in your terminal. <img alt="" width="15.994318008422852" height="15.994318008422852" src="https://avatars.githubusercontent.com/u/821668?s=80&amp;u=5e5a19b83e5d52ce7e3008a79ce0be671321efbe&amp;v=4"> @matiux嗨，你能提供plasmashell的输出吗？即直接在终端中运行 `plasmashell` 时的输出。

```shell
❯ plasmashell
kf.svg: The theme "Otto" uses the legacy metadata.desktop. Consider contacting the author and asking them update it to use the newer JSON format.
kf.svg: The theme "Otto" uses the legacy metadata.desktop. Consider contacting the author and asking them update it to use the newer JSON format.
kf.plasma.core: The theme "Otto" uses the legacy metadata.desktop. Consider contacting the author and asking them update it to use the newer JSON format.
kf.plasma.core: The theme "Otto" uses the legacy metadata.desktop. Consider contacting the author and asking them update it to use the newer JSON format.
```
